urllib3/util/retry.py

```diff
@@ -169,10 +179,13 @@ def get_backoff_time(self):
 
         :rtype: float
         """
-        if self._observed_errors <= 1:
+        # We want to consider only the last consecutive errors sequence (Ignore redirects).
+        consecutive_errors_len = len(list(takewhile(lambda x: x.redirect_location is None,
```

Comment:
>```python
consecutive_errors_len = next(i for i, h in enumerate(reversed(self.history)) if not h.redirect_location)
```

This will raise ``StopIteration`` if there aren't any redirects in the history list.

The only retry event we don't consider as error is redirect. Currently in ``get_redirect_location`` we check that ``status`` is a redirect status. The only way that ``.redirect_location`` will be different then None is if the call to ``get_redirect_location`` does not return False. IMO I think this is a simpler check then ``x.error or (x.status and x.status not in REDIRECT_STATUSES)``.